apps/server/src/schemas/comments.ts

```diff
@@ -2,14 +2,42 @@ import { z } from "zod";
 import { zDatetime } from "./common";
 import { UserSchema } from "./users";
 
+// Maximum number of mentions allowed in a single comment
+const MAX_MENTIONS = 20;
+
+// Maximum length for comment text
+const MAX_COMMENT_LENGTH = 2000;
+
 export const CommentSchema = z.object({
   id: z.number(),
-  comment: z.string(),
+  comment: z
+    .string()
+    .max(
+      MAX_COMMENT_LENGTH,
+      `Comment cannot exceed ${MAX_COMMENT_LENGTH} characters`,
+    ),
   createdAt: z.string().datetime(),
   createdBy: UserSchema,
+  replyToId: z.number().nullable().optional(),
```

Comment:
id prob make this replyTo (which is just the lazy pattern i use elsewhere in schemas)